chain/chain/src/chain.rs

```diff
@@ -2242,6 +2245,7 @@ impl Chain {
             state_patch,
         )?;
 
+        timer.observe_duration();
```

Comment:
I'd skip the `observe_duration()` call in favor of `drop()` recording the duration.

Reply:
ah didn't realize drop would record the duration too. Changed it.